src/main/drivers/dma_spi.c

```diff
@@ -0,0 +1,192 @@
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "platform.h"
+#include "dma_spi.h"
+#include "sensors/gyro.h"
+#include "drivers/accgyro/accgyro_mpu.h"
+
+
+#ifdef USE_DMA_SPI_DEVICE
+
+volatile dma_spi_read_status_t dmaSpiReadStatus = DMA_SPI_READ_UNKOWN;
+
+//must be static to avoid overflow/corruption by DMA
+uint8_t dmaTxBuffer[58];
+uint8_t dmaRxBuffer[58];
+
+static inline void gpio_write_pin(GPIO_TypeDef * GPIOx, uint16_t GPIO_Pin, uint32_t pinState)
+{
+    if (pinState != 0)
+    {
+        GPIOx->BSRRL = (uint32_t)GPIO_Pin;
+    }
+    else
+    {
+        GPIOx->BSRRH = (uint32_t)GPIO_Pin;
+    }
+}
+
+static inline void dmaSpiCsLo(void)
+{
+    gpio_write_pin(DMA_SPI_NSS_PORT, DMA_SPI_NSS_PIN, 0);
+}
+
+static inline void dmaSpiCsHi(void)
+{
+    gpio_write_pin(DMA_SPI_NSS_PORT, DMA_SPI_NSS_PIN, 1);
```

Comment:
these functions probably belong in the gpio driver files, but the bf hal is being used there, so maybe we need a new "fast_gpio" driver